tella-locking-ui/src/main/java/com/hzontal/tella_locking_ui/common/util/DivviupUtils.kt

```diff
@@ -34,7 +43,302 @@ object DivviupUtils {
                 Timber.e(e, "Divviup sending runUnlockEvent failed")
             }
         }
+    }
+
+    /**
+     * This should be called only once when user enables sending metrics
+     * Which means that the installation is done and the sending of analytics is approved
+     **/
+    fun runInstallEvent(context: Context) {
+        // If sending analytics is not accepted or install metric is sent, just return
+        if (!CommonPreferences.hasAcceptedAnalytics() || CommonPreferences.isInstallMetricSent() || isDebugBuild(context)) return
+        Executors.newSingleThreadExecutor().execute {
+            try {
+                val taskId = TaskId.parse(context.getString(R.string.divviup_count_installs_id))
+                val leaderEndpoint: URI =
+                    URI.create(context.getString(R.string.divviup_leader))
+                val helperEndpoint: URI =
+                    URI.create(context.getString(R.string.divviup_helper))
+                val timePrecisionSeconds: Long =
+                    context.resources.getInteger(R.integer.divviup_count_unlocks_timePrecisionSeconds)
+                        .toLong()
+                val client = Client.createPrio3Count(
+                    context, leaderEndpoint, helperEndpoint, taskId, timePrecisionSeconds
+                )
+                client.sendMeasurement(true)
+                CommonPreferences.setInstallMetricSent(true)
+                Timber.d("Divviup runInstallEvent measurement sent")
+            } catch (e: Exception) {
+                Timber.e(e, "Divviup sending runInstallEvent failed")
+            }
+        }
+    }
+
+    /**
+     * This task is called on Lock
+     * and sending time spent since unlock
+     **/
+    fun runTimeSpentEvent(context: Context, milliseconds: Long) {
+        Executors.newSingleThreadExecutor().execute {
```

Comment:
Remove Executors.newSingleThreadExecutor() Calls 
Consider using a Scheduler (e.g., from RxJava) or a `CoroutineDispatcher` to handle threading ...

I know that `Executors` creates a single-threaded executor, but managing the lifecycle of this executor is our  responsibility and it could be hard to do this ..

Reply:
We can use scheduler from  Rx and inject it using dagger for example : 

```
 @Provides
    @Singleton
    fun provideScheduler(): Scheduler = Schedulers.io()
```